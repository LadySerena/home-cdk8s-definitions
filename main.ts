import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";

import { SystemRbac } from "./lib/system-rbac";
import { MetricsServer } from "./lib/metrics-server";

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    // define resources here

    new SystemRbac(this, "serena-rbac", {
      user: "serena",
      resourcePrefix: "tel",
    });
    new MetricsServer(this, "metrics-server", {});
  }
}

const app = new App();
new MyChart(app, "home-kubernetes-js");
app.synth();
