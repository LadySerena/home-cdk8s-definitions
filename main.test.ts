import { CiliumChart } from "./main";
import { Testing } from "cdk8s";

describe("Placeholder", () => {
  test("Empty", () => {
    const app = Testing.app();

    const cilium = new CiliumChart(app, "foo");
    const results = Testing.synth(cilium);

    expect(results).toMatchSnapshot();
  });
});
