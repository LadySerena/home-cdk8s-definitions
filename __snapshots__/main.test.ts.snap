// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Empty 1`] = `
Array [
  Object {
    "aggregationRule": Object {
      "clusterRoleSelectors": Array [
        Object {
          "matchLabels": Object {
            "rbac.serenacodes.com/aggregate-to-cluster-admin": "true",
          },
        },
      ],
    },
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRole",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "home-lab-rbac",
      },
      "name": "tel-cluster-admin",
    },
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRole",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "home-lab-rbac",
        "rbac.serenacodes.com/aggregate-to-cluster-admin": "true",
      },
      "name": "tel-admin-core-resources",
    },
    "rules": Array [
      Object {
        "apiGroups": Array [
          "",
        ],
        "resources": Array [
          "bindings",
          "configmaps",
          "endpoints",
          "namespaces",
          "nodes",
          "persistentvolumeclaims",
          "persistentvolumes",
          "pods",
          "replicationcontrollers",
          "resourcequotas",
          "secrets",
          "serviceaccounts",
          "services",
        ],
        "verbs": Array [
          "*",
        ],
      },
      Object {
        "apiGroups": Array [
          "",
        ],
        "resources": Array [
          "events",
        ],
        "verbs": Array [
          "get",
          "list",
          "watch",
        ],
      },
      Object {
        "apiGroups": Array [
          "apiextensions.k8s.io",
        ],
        "resources": Array [
          "customresourcedefinitions",
        ],
        "verbs": Array [
          "*",
        ],
      },
      Object {
        "apiGroups": Array [
          "apiregistration.k8s.io",
        ],
        "resources": Array [
          "apiservices",
        ],
        "verbs": Array [
          "*",
        ],
      },
      Object {
        "apiGroups": Array [
          "authentication.k8s.io",
        ],
        "resources": Array [
          "localsubjectaccessreviews",
          "selfsubjectaccessreviews",
          "selfsubjectrulesreviews",
          "subjectaccessreviews",
        ],
        "verbs": Array [
          "*",
        ],
      },
      Object {
        "apiGroups": Array [
          "batch",
        ],
        "resources": Array [
          "cronjobs",
          "jobs",
        ],
        "verbs": Array [
          "*",
        ],
      },
      Object {
        "apiGroups": Array [
          "apps",
        ],
        "resources": Array [
          "controllerrevisions",
          "daemonsets",
          "replicasets",
          "statefulsets",
        ],
        "verbs": Array [
          "*",
        ],
      },
      Object {
        "apiGroups": Array [
          "autoscaling",
        ],
        "resources": Array [
          "horizontalpodautoscalers",
        ],
        "verbs": Array [
          "*",
        ],
      },
      Object {
        "apiGroups": Array [
          "events.k8s.io",
        ],
        "resources": Array [
          "events",
        ],
        "verbs": Array [
          "get",
          "list",
          "watch",
        ],
      },
      Object {
        "apiGroups": Array [
          "networking.k8s.io",
        ],
        "resources": Array [
          "ingressclasses",
          "ingresses",
          "networkpolicies",
        ],
        "verbs": Array [
          "*",
        ],
      },
      Object {
        "apiGroups": Array [
          "policy",
        ],
        "resources": Array [
          "poddisruptionbudgets",
        ],
        "verbs": Array [
          "*",
        ],
      },
      Object {
        "apiGroups": Array [
          "rbac.authorization.k8s.io",
        ],
        "resources": Array [
          "clusterrolebindings",
          "clusterroles",
          "rolebindings",
          "roles",
        ],
        "verbs": Array [
          "*",
        ],
      },
    ],
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRoleBinding",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "home-lab-rbac",
      },
      "name": "tel-serena-rolebinding",
    },
    "roleRef": Object {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "ClusterRole",
      "name": "tel-cluster-admin",
    },
    "subjects": Array [
      Object {
        "apiGroup": "rbac.authorization.k8s.io",
        "kind": "User",
        "name": "serena",
      },
    ],
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "metrics-server",
      },
      "name": "metrics-server",
      "namespace": "kube-system",
    },
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRole",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "metrics-server",
        "rbac.authorization.k8s.io/aggregate-to-admin": "true",
        "rbac.authorization.k8s.io/aggregate-to-edit": "true",
        "rbac.authorization.k8s.io/aggregate-to-view": "true",
      },
      "name": "system:metrics-server-aggregated-reader",
    },
    "rules": Array [
      Object {
        "apiGroups": Array [
          "metrics.k8s.io",
        ],
        "resources": Array [
          "pods",
          "nodes",
        ],
        "verbs": Array [
          "get",
          "list",
          "watch",
        ],
      },
    ],
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRole",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "metrics-server",
      },
      "name": "system:metrics-server",
    },
    "rules": Array [
      Object {
        "apiGroups": Array [
          "",
        ],
        "resources": Array [
          "nodes/metrics",
        ],
        "verbs": Array [
          "get",
        ],
      },
      Object {
        "apiGroups": Array [
          "",
        ],
        "resources": Array [
          "pods",
          "nodes",
          "namespaces",
          "configmaps",
        ],
        "verbs": Array [
          "get",
          "list",
          "watch",
        ],
      },
    ],
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRoleBinding",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "metrics-server",
      },
      "name": "metrics-server:system:auth-delegator",
    },
    "roleRef": Object {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "ClusterRole",
      "name": "system:auth-delegator",
    },
    "subjects": Array [
      Object {
        "kind": "ServiceAccount",
        "name": "metrics-server",
        "namespace": "kube-system",
      },
    ],
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRoleBinding",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "metrics-server",
      },
      "name": "system:metrics-server",
    },
    "roleRef": Object {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "ClusterRole",
      "name": "system:metrics-server",
    },
    "subjects": Array [
      Object {
        "kind": "ServiceAccount",
        "name": "metrics-server",
        "namespace": "kube-system",
      },
    ],
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "RoleBinding",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "metrics-server",
      },
      "name": "metrics-server-auth-reader",
      "namespace": "kube-system",
    },
    "roleRef": Object {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "Role",
      "name": "extension-apiserver-authentication-reader",
    },
    "subjects": Array [
      Object {
        "kind": "ServiceAccount",
        "name": "metrics-server",
        "namespace": "kube-system",
      },
    ],
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "metrics-server",
      "namespace": "kube-system",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "name": "https",
          "port": 443,
          "protocol": "TCP",
          "targetPort": "https",
        },
      ],
      "selector": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "metrics-server",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "metrics-server",
      },
      "name": "metrics-server",
      "namespace": "kube-system",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app.kubernetes.io/managed-by": "cdk8s",
          "app.kubernetes.io/name": "metrics-server",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app.kubernetes.io/managed-by": "cdk8s",
            "app.kubernetes.io/name": "metrics-server",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "args": Array [
                "--secure-port=4443",
                "--cert-dir=/tmp",
                "--kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname",
                "--kubelet-use-node-status-port",
                "--metric-resolution=30s",
                "--kubelet-insecure-tls",
              ],
              "image": "k8s.gcr.io/metrics-server/metrics-server:v0.6.1",
              "imagePullPolicy": "IfNotPresent",
              "livenessProbe": Object {
                "failureThreshold": 4,
                "httpGet": Object {
                  "path": "/livez",
                  "port": "https",
                  "scheme": "HTTPS",
                },
                "initialDelaySeconds": 15,
                "periodSeconds": 10,
              },
              "name": "metrics-server",
              "ports": Array [
                Object {
                  "containerPort": 4443,
                  "name": "https",
                  "protocol": "TCP",
                },
              ],
              "readinessProbe": Object {
                "failureThreshold": 3,
                "httpGet": Object {
                  "path": "/readyz",
                  "port": "https",
                  "scheme": "HTTPS",
                },
                "initialDelaySeconds": 20,
                "periodSeconds": 10,
              },
              "securityContext": Object {
                "allowPrivilegeEscalation": false,
                "readOnlyRootFilesystem": true,
                "runAsNonRoot": true,
                "runAsUser": 1000,
              },
              "volumeMounts": Array [
                Object {
                  "mountPath": "/tmp",
                  "name": "tmp",
                },
              ],
            },
          ],
          "priorityClassName": "system-cluster-critical",
          "serviceAccountName": "metrics-server",
          "volumes": Array [
            Object {
              "emptyDir": Object {},
              "name": "tmp",
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "apiregistration.k8s.io/v1",
    "kind": "APIService",
    "metadata": Object {
      "labels": Object {
        "app.kubernetes.io/managed-by": "cdk8s",
        "app.kubernetes.io/name": "metrics-server",
      },
      "name": "v1beta1.metrics.k8s.io",
    },
    "spec": Object {
      "group": "metrics.k8s.io",
      "groupPriorityMinimum": 100,
      "insecureSkipTLSVerify": true,
      "service": Object {
        "name": "metrics-server",
        "namespace": "kube-system",
      },
      "version": "v1beta1",
      "versionPriority": 100,
    },
  },
]
`;
